styx-common/src/test/java/com/spotify/styx/util/TimeUtilTest.java

```diff
@@ -187,53 +191,149 @@ public void shouldAddOffsetWithWeek() {
-               is(9));
+    assertThat(instantsInRange(firstTimeHours, lastTimeHours, Schedule.parse("0 * * * *")),
+        contains(parse("2016-01-19T00:00:00.00Z"),
+            parse("2016-01-19T01:00:00.00Z"),
+            parse("2016-01-19T02:00:00.00Z")));
+  }
+
+  @Test
+  public void shouldReturnEmptyList() {
+    final Instant firstTimeHours = parse("2016-01-19T00:00:00.00Z");
+    final Instant lastTimeHours = parse("2016-01-19T00:00:00.00Z");
+
+    assertTrue(
+        instantsInRange(firstTimeHours, lastTimeHours, Schedule.parse("0 * * * *")).isEmpty());
   }
 
   @Test(expected = IllegalArgumentException.class)
   public void shouldGetExceptionIfLastInstantIsBeforeFirstInstant() {
-    final Instant firstTimeHours = Instant.parse("2016-01-19T10:00:00.00Z");
-    final Instant lastTimeHours = Instant.parse("2016-01-19T09:00:00.00Z");
-    instancesInRange(firstTimeHours, lastTimeHours, Schedule.HOURS);
+    final Instant firstTimeHours = parse("2016-01-19T10:00:00.00Z");
+    final Instant lastTimeHours = parse("2016-01-19T09:00:00.00Z");
+    instantsInRange(firstTimeHours, lastTimeHours, Schedule.HOURS);
   }
 
   @Test(expected = IllegalArgumentException.class)
   public void shouldGetExceptionIfLastInstantIsNotAlignedWithSchedule() {
-    final Instant firstTimeHours = Instant.parse("2016-01-19T08:00:00.00Z");
-    final Instant lastTimeHours = Instant.parse("2016-01-19T09:10:00.00Z");
-    instancesInRange(firstTimeHours, lastTimeHours, Schedule.HOURS);
+    final Instant firstTimeHours = parse("2016-01-19T08:00:00.00Z");
+    final Instant lastTimeHours = parse("2016-01-19T09:10:00.00Z");
+    instantsInRange(firstTimeHours, lastTimeHours, Schedule.HOURS);
   }
 
   @Test(expected = IllegalArgumentException.class)
   public void shouldGetExceptionIfFirstInstantIsNotAlignedWithSchedule() {
-    final Instant firstTimeHours = Instant.parse("2016-01-19T08:10:00.00Z");
-    final Instant lastTimeHours = Instant.parse("2016-01-19T09:00:00.00Z");
-    instancesInRange(firstTimeHours, lastTimeHours, Schedule.HOURS);
+    final Instant firstTimeHours = parse("2016-01-19T08:10:00.00Z");
+    final Instant lastTimeHours = parse("2016-01-19T09:00:00.00Z");
+    instantsInRange(firstTimeHours, lastTimeHours, Schedule.HOURS);
+  }
+
+  @Test
+  public void shouldGetCorrectNumberOfInstantsReversed() {
+    final Instant firstTimeHours = parse("2016-01-19T03:00:00.00Z");
+    final Instant lastTimeHours = parse("2016-01-19T00:00:00.00Z");
+    final Instant firstTimeDays = parse("2016-01-13T00:00:00.00Z");
+    final Instant lastTimeDays = parse("2016-01-10T00:00:00.00Z");
+    final Instant firstTimeWeeks = parse("2016-01-18T00:00:00.00Z");
+    final Instant lastTimeWeeks = parse("2016-01-11T00:00:00.00Z");
+    final Instant firstTimeMonths = parse("2013-01-01T00:00:00.00Z");
+    final Instant lastTimeMonths = parse("2010-01-01T00:00:00.00Z");
+
+    assertThat(instantsInReversedRange(firstTimeHours, lastTimeHours, Schedule.HOURS),
```

Comment:
Is it expected that `instantsInRange(x, y, ...)` and `instantsInReversedRange(y, x, ...)` do not produce the same set of instants?

If the user supplies the same start/end timestamps to the backfill create command but in reverse, should it not produce the exact same workflow instances but just in reverse?

Reply:
It is expected due to inclusive (x) and exclusive (y).

Reply:
Yeah, and I think that might be quite confusing for the user. Might it be easier to use if `styx backfill create <component> <workflow> 2017-01-01 2018-01-01 <concurrency>` and 
`styx backfill create <component> <workflow> 2018-01-01 2017-01-01 <concurrency>` result in the same partitions, just the order reversed?

Reply:
I think it might be OK because we have been explicitly stating the `end` is exclusive, so unless we want to make both of them inclusive for normal and reversed order, it might be more confusing.

Reply:
Maybe this needs some user testing.

Reply:
The more I think about this, the more it seems like having an explicit `reverse` flag might be better, both for humans using the cli and for machines calling the api.

E.g. if we add backfill creation to System-Z and then have a `reverse` check box that works by swapping the dates and shifting them 1 partition in one direction, that seems awkward to me. Then I would rather just have a `reverse=true|false` parameter.

And `styx backfill create --reverse ...` seems more clear than swapping start/end.